boards/risc-v/mpfs/icicle/include/board_memorymap.h

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * boards/risc-v/mpfs/icicle/include/board_memorymap.h
```

Comment:
What about `m100pfsevp` board? Does it need the same file as well?

Reply:
This is not strictly required at all. It is for convenience.

The reason I made this file is that CONFIG_BUILD_KERNEL code will require a memory map, and the linker provided symbols need to be accessible from multiple files. Just to avoid copy&pasting the "extern __foobar" to each file, I made this file which can be used to access the symbols.

Reply:
But you should probably wrap `#include <arch/board/board_memorymap.h>` with `#ifdef CONFIG_MM_KERNEL_HEAP`?

Reply:
I can of course do that

Reply:
I expect that otherwise build should fail

Reply:
So either `ifdef CONFIG_MM_KERNEL_HEAP` or create an empty `board_memorymap.h` for `m100pfsevp` board

Reply:
Absolutely true, did not even realize there was another mpfs board target.